src/wild_encounter.c

```diff
@@ -0,0 +1,748 @@
+{
+    u16 i;
+
+    for (i = 0; ; i++)
+    {
+        const struct WildPokemonHeader *wildHeader = &gWildMonHeaders[i];
+        if (wildHeader->mapGroup == 0xFF)
+            break;
+
+        if (gWildMonHeaders[i].mapGroup == gSaveBlock1Ptr->location.mapGroup &&
+            gWildMonHeaders[i].mapNum == gSaveBlock1Ptr->location.mapNum)
+        {
+            if (gSaveBlock1Ptr->location.mapGroup == MAP_GROUP(SIX_ISLAND_ALTERING_CAVE) &&
+                gSaveBlock1Ptr->location.mapNum == MAP_NUM(SIX_ISLAND_ALTERING_CAVE))
+            {
+                u16 alteringCaveId = VarGet(VAR_ALTERING_CAVE_WILD_SET);
+                if (alteringCaveId > 8)
+                    alteringCaveId = 0;
+
+                i += alteringCaveId;
+            }
+
+            if (!UnlockedTanobyOrAreNotInTanoby())
+                break;
+            return i;
+        }
+    }
+
+    return -1;
+}
+
+static bool8 UnlockedTanobyOrAreNotInTanoby(void)
+{
+    if (FlagGet(FLAG_SYS_UNLOCKED_TANOBY_RUINS))
+        return TRUE;
+    if (gSaveBlock1Ptr->location.mapGroup != MAP_GROUP(SEVEN_ISLAND_TANOBY_RUINS_DILFORD_CHAMBER))
+        return TRUE;
+    if (!(gSaveBlock1Ptr->location.mapNum == MAP_NUM(SEVEN_ISLAND_TANOBY_RUINS_MONEAN_CHAMBER)
+    ||  gSaveBlock1Ptr->location.mapNum == MAP_NUM(SEVEN_ISLAND_TANOBY_RUINS_LIPTOO_CHAMBER)
+    ||  gSaveBlock1Ptr->location.mapNum == MAP_NUM(SEVEN_ISLAND_TANOBY_RUINS_WEEPTH_CHAMBER)
+    ||  gSaveBlock1Ptr->location.mapNum == MAP_NUM(SEVEN_ISLAND_TANOBY_RUINS_DILFORD_CHAMBER)
+    ||  gSaveBlock1Ptr->location.mapNum == MAP_NUM(SEVEN_ISLAND_TANOBY_RUINS_SCUFIB_CHAMBER)
+    ||  gSaveBlock1Ptr->location.mapNum == MAP_NUM(SEVEN_ISLAND_TANOBY_RUINS_RIXY_CHAMBER)
+    ||  gSaveBlock1Ptr->location.mapNum == MAP_NUM(SEVEN_ISLAND_TANOBY_RUINS_VIAPOIS_CHAMBER)
+    ))
+        return TRUE;
+    return FALSE;
+}
+
+static void GenerateWildMon(u16 species, u8 level, u8 slot)
+{
+    u32 personality;
+    s8 chamber;
+    ZeroEnemyPartyMons();
+    if (species != SPECIES_UNOWN)
+    {
+        CreateMonWithNature(&gEnemyParty[0], species, level, 32, Random() % 25);
+    }
+    else
+    {
+        chamber = gSaveBlock1Ptr->location.mapNum - MAP_NUM(SEVEN_ISLAND_TANOBY_RUINS_MONEAN_CHAMBER);
+        personality = GenerateUnownPersonalityByLetter(gUnknown_83CA71C[chamber][slot]);
+        CreateMon(&gEnemyParty[0], species, level, 32, TRUE, personality, FALSE, 0);
+    }
+}
+
+static u32 GenerateUnownPersonalityByLetter(u8 letter)
+{
+    u32 personality;
+    do
+    {
+        personality = (Random() << 16) | Random();
+    } while (GetUnownLetterByPersonalityLoByte(personality) != letter);
+    return personality;
+}
+
+u8 GetUnownLetterByPersonalityLoByte(u32 personality)
+{
+    return (((personality & 0x3000000) >> 18) | ((personality & 0x30000) >> 12) | ((personality & 0x300) >> 6) | (personality & 0x3)) % 0x1C;
+}
+
+static bool8 TryGenerateWildMon(const struct WildPokemonInfo * info, u8 tableIdx, u8 a2)
```

Comment:
a2 might be bool8 as it's set to 0 only when in sweetscent encounter (else always 1)

Reply:
Nope, it's the check flags from emerald.